README.md

```diff
@@ -4,29 +4,127 @@
 
 This is a Node module that makes it very simple
 to interact with MySQL databases.
-It has the same API as https://github.com/mvolkmann/postgresql-easy.
+Fundamentally, it wraps Connection and Pool objects from [mysql](https://www.npmjs.com/package/mysql) and adds convenience methods for common tasks and interaction patterns.
+It was inspired by [postgres-easy](https://github.com/mvolkmann/postgresql-easy).
 
-To install this, run `npm install -S mysql-easier`
+Migrating from 1.x to 2.x?  Please see: [Migrating](#migrating-from-1x-to-2x)
 
-## Setup
+## Installation
+
+To install mysql-easier, run `npm install -S mysql-easier`
+
+## General Use
 
 ```js
-const MySqlConnection = require('mysql-easier');
-const mysql = new MySqlConnection(config);
+const mySqlEasier = require('mysql-easier');
+
+// Configure the global pool
+mySqlEasier.configure(config);
+
+// Get a connection from the pool
+const myConn = await mySqlEasier.getConnection();
+
+// Do some database stuff
+
+const newId = await myConn.insert('food', {name: 'Orange', color: 'Orange'});
+
+const aFruit = await myConn.getById('food', newId);
+console.log(aFruit.name) // Output: Orange
+
+await myConn.deleteById('food', newId);
+
+// All done?  Release the connection
+await myConn.done();
+
+// App ready to exit?  Close the pool
+await mySqlEasier.endPool();
 ```
 
-The config object can contain these properties:
-* `acquireTimeout`: time before the connection is closed; default is 30000
-* `connectionLimit`: maximum number of concurrent connections; default is 10
-* `database`: the name of the database to use
-* `debug`: true to output messages describing each action; defaults to false
-* `host`: defaults to localhost
-* `password`: if database requires authentication
-* `queueLimit`: maximum number of queued connection requests; default is 0
-* `user`: if database requires authentication
-* `waitForConnections`: boolean that determines if connection requests should be queued if there are no available connections; default is true
+## Pool and Connection Management
+
+Let's break down the management of connections in mysql-easier.
+Your index or main script in your application will likely configure your database pool:
+
+```js
+// Index or Main module
+const mySqlEasier = require('mysql-easier');
+
+mySqlEasier.configure(config);
+```
+
+At this point, mysql-easier will have created a global connection pool for your database interactions using the configuration provided.  To get a connection from the pool (including from other modules):
+
+```js
+// Some other module
+const mySqlEasier = require('mysql-easier');
+
+const myConn = await mySqlEasier.getConnection();
+
+// Do some database stuff
+
+await myConn.done(); // Releases the connection back to the pool.
+
+// --OR--
```

Comment:
I think it's okay to only show usage with async/await. I think the style of using the Promise then method it gradually fading away from common practice.

Reply:
OK.  I'll remove that part.